src/routes/object/createObject.ts

```diff
@@ -237,6 +248,47 @@ export default async function routes(fastify: FastifyInstance) {
       return response.status(uploadResult.httpStatusCode ?? 200).send({
         Key: path,
       })
+
+      /**
+       * Remove row from `storage.objects` if there was an error uploading to the backend.
+       */
+      async function handleUploadError(uploadError: unknown) {
+        if (uploadError) {
+          request.log.error({ error: uploadError }, 'upload error object')
+
+          let errorName: string
+          let errorStatusCode: number
+          let errorMessage: string | undefined
+
+          if (isS3Error(uploadError)) {
+            errorName = uploadError.name
```

Comment:
Lets normalise the errors before returning. We shouldn't leak that the backend is s3, file or backblaze, etc to the end user.